x/blockdb/memory_db.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blockdb
+
+import "sync"
+
+// MemoryDatabase is an in-memory implementation of BlockDB
+type MemoryDatabase struct {
+	mu     sync.RWMutex
+	blocks map[BlockHeight]BlockData
+	closed bool
+}
+
+// NewMemoryDatabase creates a new in-memory BlockDB
+func NewMemoryDatabase() *MemoryDatabase {
+	return &MemoryDatabase{
+		blocks: make(map[BlockHeight]BlockData),
+	}
+}
+
+// WriteBlock stores a block in memory
+func (m *MemoryDatabase) WriteBlock(height BlockHeight, block BlockData) error {
```

Comment:
blockData is a slice so why do we copy it and not just save the reference? This is for tests. Do we expect the original slice to be changed?